src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs

```diff
@@ -29,16 +31,39 @@ public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(VisualStudioWorksp
             // Get the analyzer assets for installed VSIX extensions through the VSIX extension manager.
             var extensionManager = workspace.GetVsService<SVsExtensionManager, IVsExtensionManager>();
 
-            var builder = ImmutableArray.CreateBuilder<HostDiagnosticAnalyzerPackage>();
-            foreach (var extension in extensionManager.GetEnabledExtensions(AnalyzerContentTypeName))
+            // get rootfolder and shellfolder location
+            string rootFolder;
+            string shellFolder;
+            if (TryGetRootAndShellFolder(extensionManager, out shellFolder, out rootFolder))
```

Comment:
well that approach only uses all supported shell APIs. Relative paths for all contents of extensions are suppose to give the Path string embedded in extensions VSIX manifest, which must end in assembly/file name. Similarly content locations returned by GetEnabledExtensionContentLocations must also end in assembly/file name. Either way its not a clean approach :)

Personally, I would prefer either (a) not doing anything here and going back to the old code, until shell team exposes API for this OR (b) have workaround which uses shell APIs with supported semantics. I'd let other people on the PR comment to get their views though.